syntax = "proto3";

package course_category;


service CourseCategoryService {
    rpc GetCategory(GetCategoryRequest) returns (GetCategoryResponse);
    rpc ListCategories(ListCategoriesRequest) returns (ListCategoriesResponse);
      rpc GetCategoriesByIds(GetCategoriesByIdsRequest)
      returns (GetCategoriesByIdsResponse);
}

message GetCategoryRequest {
    string id = 1;
}

message GetCategoryResponse {
    string id = 1;
    string title = 2;
    string codename = 3;
    optional string parent_id = 4;
}

message ListCategoriesRequest {
    int32 limit = 1;
    int32 offset = 2;
}

message CategoryItem {
    string id = 1;
    string title = 2;
    string codename = 3;
    optional string parent_id = 4;
}

message ListCategoriesResponse {
    repeated CategoryItem items = 1;
    int32 total = 2;
}

message GetCategoriesByIdsRequest {
  repeated string ids = 1;
}

message GetCategoriesByIdsResponse {
  repeated CategoryItem items = 1;
}